packages/labs/src/tag-input/tagInput.tsx

```diff
@@ -19,8 +19,11 @@ export interface ITagInputProps extends IProps {
      * Callback invoked when a new tag is added by the user pressing `enter` on the input.
      * Receives the current value of the input field. New tags are expected to be appended to
      * the list.
+     *
+     * If the provided function returns `false`, the input will not be cleared. This is useful,
+     * for example, if the provided `value` is somehow invalid and should not be added as a tag.
      */
-    onAdd?: (value: string) => void;
+    onAdd?: (value: string) => boolean;
```

Comment:
What about `boolean | void`, so that you can return both `true` or nothing if you want the tag to be added?